### PR TITLE
create method problem in class group, post not allowed

### PR DESCRIPTION
**Pull request by tdohany**
create method problem in class group, post not allowed
requests.exceptions.HTTPError: 405 Client Error: Method Not Allowed for url

### CODE REVIEW

File: dohq_artifactory/admin.py [hunk 1]
```diff
@@ -501,8 +501,8 @@ def create(self):
         )
         data_json = self._create_json()
         data_json.update(self.additional_params)
-        request_url = f"{self.base_url}/{self.prefix_uri}/{self._uri}"
-        r = self._session.post(
+        request_url = f"{self.base_url}/{self.prefix_uri}/{self._uri}/{getattr(self, self.resource_name)}"
+        r = self._session.put(
```
Comment (zhan9san) [inline dohq_artifactory/admin.py:505]: Hi @tdohany 
Glad you are interested in this project.

How about removing `delete` method in Group class?
The `delete` is the same as one in `AdminObject` class.

Comment (tdohany) [reply dohq_artifactory/admin.py:505]: Hi @zhan9san 
The delete works properly, it has the same request_url like in my request and of course delete method.
request_url = f"{self.base_url}/{self.prefix_uri}/{self._uri}/{getattr(self, self.resource_name)}"

I've used this test code (GroupMy is an override the Groupl) :

# Remove if exist
test_group = artifactory.find_group(name)
if test_group is not None:
    test_group.delete()

test_group = GroupMy(artifactory=artifactory, name=name)
# CREATE
test_group.create()

test_group = artifactory.find_group(name)
if test_group is not None:
    test_group.delete()

Comment (zhan9san) [reply dohq_artifactory/admin.py:505]: Sorry for the typo.

I'd like to say removing `create` method in Group. After we introduce this change, the `create` method in Group would be the same as it in parent class `AdminObject`

Comment (tdohany) [reply dohq_artifactory/admin.py:505]: Basically yes. But why do we need a class group method create if it doesn't work (neither In the tests)

Comment (zhan9san) [reply dohq_artifactory/admin.py:505]: it worked before

and it will be great if you could add some unit tests

Comment (tdohany) [reply dohq_artifactory/admin.py:505]: I think it should be tested by integration or e2e test.

Comment (zhan9san) [reply dohq_artifactory/admin.py:505]: IMHO, there is no conflict between integration test and unit test.
If you are interested in it, there are [some examples](https://github.com/devopshq/artifactory/blob/master/tests/unit/test_artifactory_path.py#L1278-L1330) in unit test file for your reference.

Comment (tdohany) [reply dohq_artifactory/admin.py:505]: I understand, it is working as it written and expected but see the e2e test attached. These are a requests/responses from the real system.
[tests.txt](https://github.com/devopshq/artifactory/files/7234274/tests.txt)

Comment (allburov) [reply dohq_artifactory/admin.py:505]: We can postpone refactoring part

**Pull request merged**